Yasgui/packages/yasgui/src/index.ts

```diff
@@ -189,31 +189,50 @@ export class Yasgui extends EventEmitter {
         if (executeIdAfterInit && executeIdAfterInit === activeTabId) {
           (this.getTab(activeTabId) as Tab).query().catch(() => {});
         }
-        // }
       }
     }
   }
 
   public getHandleLocalStorageQuotaFull(): (e: any) => void {
     return () => {
-      this.resetResults();
+      this.reInitYasgui({resetYasr: true});
     };
   }
 
-  public resetResults(resetCurrentTab = false) {
+    /**
+     * Reinitializes the YASGUI component with default values from the config.
+     * @param {YasguiResetFlags} resetFlags object defining the flags for what to reset
+     * @param {YasguiConfiguration} config the YasguiConfiguration used to get the default values
+     */
+  public reInitYasgui(resetFlags: any, config: any = {}) {
     const currentId = this.persistentConfig.currentId();
+    const refreshYasqe = this.hasYasqeRefresh(resetFlags.infer, config?.yasguiConfig?.infer);
     Object.values(this.persistentConfig.getTabConfig()).forEach((tab: any) => {
-      if (resetCurrentTab || currentId !== tab.id) {
-        tab.yasr.response = null;
+      if (resetFlags.resetCurrentTab || currentId !== tab.id) {
+        if (resetFlags.resetYasr) {
+          tab.yasr.response = null;
+        }
+      }
+      if (refreshYasqe) {
+          tab.yasqe.infer = config.yasguiConfig.infer;
+          tab.yasqe.sameAs = config.yasguiConfig.sameAs || config.yasguiConfig.infer;
       }
     });
     Object.values(this._tabs).forEach((tab: Tab) => {
-      if (resetCurrentTab || currentId !== tab.getId()) {
+      if (resetFlags.resetCurrentTab || currentId !== tab.getId()) {
         this.emitTabChange(tab);
         tab.reInitYasr();
+          if (refreshYasqe) {
+              tab.getYasqe()?.refresh();
+          }
       }
     });
   }
+
+  public hasYasqeRefresh(resetInfer: any, configInfer: any) {
```

Comment:
This don't need to be public

Reply:
Also, stop using `any` when there is a matching type. Or if there isn't, then create one!

Reply:
And another one. What is the meaning of this method name? I don't get the between this name and the implementation.